apps/tx-builder/src/components/TransactionBatchListItem.tsx

```diff
@@ -0,0 +1,352 @@
+                        <TransactionActionButton
+                          size="medium"
+                          aria-label="Edit transaction"
+                          onClick={(event) => {
+                            event.stopPropagation();
+                            setTxIndexToEdit(String(index));
+                            openEditTxModal();
+                          }}
+                        >
+                          <Icon size="sm" type="edit" />
+                        </TransactionActionButton>
+                      </Tooltip>
+                    )}
+
+                    {/* Delete transaction */}
+                    {removeTransaction && (
+                      <Tooltip
+                        placement="top"
+                        title="Delete transaction"
+                        backgroundColor="primary"
+                        textColor="white"
+                        arrow
+                      >
+                        <TransactionActionButton
+                          onClick={(event) => {
+                            event.stopPropagation();
+                            setTxIndexToRemove(String(index));
+                            openDeleteTxModal();
+                          }}
+                          size="medium"
+                          aria-label="Delete transaction"
+                        >
+                          <Icon size="sm" type="delete" />
+                        </TransactionActionButton>
+                      </Tooltip>
+                    )}
+
+                    {/* Expand transaction details */}
+                    {showTransactionDetails && (
+                      <Tooltip
+                        placement="top"
+                        title="Expand transaction details"
+                        backgroundColor="primary"
+                        textColor="white"
+                        arrow
+                      >
+                        <TransactionActionButton
+                          onClick={(event) => {
+                            event.stopPropagation();
+                            onClickShowTransactionDetails();
+                          }}
+                          size="medium"
+                          aria-label="Expand transaction details"
+                        >
+                          <FixedIcon type={'chevronDown'} />
+                        </TransactionActionButton>
+                      </Tooltip>
+                    )}
+                  </AccordionSummary>
+                </div>
+
+                {/* Transaction details */}
+                <AccordionDetails>
+                  <TransactionDetails transaction={transaction} />
+                </AccordionDetails>
+              </StyledAccordion>
+            </TransactionListItem>
+          );
+        }}
+      </Draggable>
+    );
+  },
+);
+
+export const getTransactionText = (description: ProposedTransaction['description']) => {
```

Comment:
would move it to a file with utils, the component is already big enough without additional logic, plus it is exported and used in other components

Reply:
Good idea